content/docs/04-clusters/03-edge/02-edge-configuration/30-cloud-init.md

```diff
@@ -59,6 +59,60 @@ You may ask yourself where to use cloud-init stages, as both the Edge Installer
 
 * If you have common configurations across a fleet of Edge host devices, customize the OS pack and use the cloud-init stages to apply those configurations. 
 
+# Sensitive Information in the User Data Stages
+
+Suppose you must add sensitive information, such as credentials, in your user data configuration file. In the Edge deployment lifecycle, you have two opportunities to apply user data to edge hosts. The first is during the staging phase, where you add the Edge installer to the Edge host. The second opportunity is during the site installation phase, where you can provide supplementary user-data configurations if needed. The diagram below highlights the two mentioned phases in the Edge lifecycle.  
+
+![A diagram highlighting the two stages in the edge deployment lifecycle where you can apply user data.](/edge_edge-configuration_cloud-init_user-data.png)
+<br />
+
+<br />
+
+- **Staging Phase** - In the staging phase, you prepare your edge hosts using the organization-level configurations. The configurations include the Edge Installer, the user data, and, optionally, a content bundle. You boot the edge hosts using the Edge Installer and apply the configurations. All the configurations, including the user data, are copied to the edge host during installation. 
+
+  Once the edge hosts are prepared with the initial installation, you ship your devices to the site for installation. This step is also called the *installer handoff* step. Refer to the [Prepare Edge Host
+](/clusters/edge/site-deployment/stage#prepareedgehost) guide to learn more about driving the installer handoff step.
+
+
+- **Site Installation Phase** - In the site installation phase, you use supplementary user data to apply site-specific configurations to the edge devices. The user data is copied to the edge host during the installation unless you follow the specific naming convention for your user data stages as described below. 
+
+ The [Perform Site Install](/clusters/edge/site-deployment/site-installation) guide describes the site installation process in detail. If you need to apply a supplementary user-data configuration file,  refer to the [Multiple User Data Use Case](/clusters/edge/edgeforge-workflow/prepare-user-data#multipleuserdatausecase) guide to learn more.
+
+
+In both steps mentioned above,  the Edge Installer copies the user data configuration file provided to the **/run/stylus/userdata** file or the **/oem/userdata** file on the edge hosts. If you want to prevent some user data stages from getting copied to the edge host's storage, you can use a specific naming convention to disable the default copy behavior. However, be aware that different persistence behaviors apply depending on which stage of the Edge deployment life cycle you provide sensitive data in the user data configuration file. Refer to the [Sensitive Information in the Site Installation](#sensitiveinformationduringthesiteinstallation) section below to learn more.
```

Comment:
```suggestion
In both steps mentioned above, the Edge Installer copies the user data configuration file provided to the **/run/stylus/userdata** file or the **/oem/userdata** file on the edge hosts. If you want to prevent some user data stages from getting copied to the edge host's storage, you can use a specific naming convention to disable the default copy behavior. However, be aware that different persistence behaviors apply depending on which stage of the Edge deployment life cycle you provide sensitive data in the user data configuration file. Refer to the [Sensitive Information in the Site Installation](#sensitiveinformationduringthesiteinstallation) section below to learn more.
```